### PR TITLE
GCE cloud api functions needed for l7 controller

### PR DESCRIPTION
**Pull request by bprashanth**
Basic accessor methods for resources that build up to an l7 loadbalancer on gce. See https://github.com/kubernetes/kubernetes/pull/12825 for context.

### CODE REVIEW

**Comment by bprashanth** [comment]
@a-robinson can you review/recommend a reviewer?

File: pkg/cloudprovider/providers/gce/gce.go [hunk 1]
```diff
@@ -566,6 +566,293 @@ func (gce *GCECloud) EnsureTCPLoadBalancerDeleted(name, region string) error {
...
 
+// UrlMap management
+
+// GetUrlMap returns the UrlMap by name.
+func (gce *GCECloud) GetUrlMap(name string) (*compute.UrlMap, error) {
+	return gce.service.UrlMaps.Get(gce.projectID, name).Do()
+}
+
+// CreateUrlMap creates an url map, using the given backend service as the default service.
+func (gce *GCECloud) CreateUrlMap(backend *compute.BackendService, name string) (*compute.UrlMap, error) {
```
Comment (cjcullen) [inline pkg/cloudprovider/providers/gce/gce.go]: If we want this to be a generic method in the cloudprovider Interface (do we?), it shouldn't take/return compute pkg structs. But I suppose we can cross that bridge when we come to it.

**Comment by cjcullen** [comment]
LGTM as a launching pad for building out L7 support.

It does highlight the fact that the single gce.go file is getting very cluttered, but no need to tackle that in this PR.

**Comment by cjcullen** [comment]
@k8s-bot test this please

**Comment by a-robinson** [comment]
Sorry for being a little slow getting to this, @bprashanth! Thanks for reviewing, @cjcullen.

File: pkg/cloudprovider/providers/gce/gce.go (hunk 1 above)
Comment (bprashanth) [inline pkg/cloudprovider/providers/gce/gce.go]: You mean like Route does right? https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/cloud.go#L119
I didn't want to impose the concept of urlmap/targetproxy/instance groups on all cloudproviders. At this point it'll be boilerplate for the other providers. I'll revist this when I figure out what these concepts map to on the other providers.

Comment (cjcullen) [inline pkg/cloudprovider/providers/gce/gce.go]: I think that's the right approach.

**Comment by bprashanth** [comment]
@k8s-bot test this please

**Comment by bprashanth** [comment]
Previous failures were jenkins disk space issues

**Pull request merged**